src/server/context.ts

```diff
@@ -103,7 +105,32 @@ export class ServerContext {
   ): Promise<ServerContext> {
     // Get the manifest' base URL.
     const baseUrl = new URL("./", manifest.baseUrl).href;
-    const importMapURL = new URL("./import_map.json", manifest.baseUrl);
+
+    const config = manifest.config || { importMap: "./import_map.json" };
+    if (typeof config.importMap !== "string") {
+      throw new Error("deno.json must contain an 'importMap' property.");
+    }
+    const importMapURL = new URL(config.importMap, manifest.baseUrl);
+
+    config.compilerOptions ??= {};
```

Comment:
Incomplete check?

It seems this missing check and the usage of `config.compilerOptions.jsx` below create an implicit dependency on the existence of a `./deno.json`. Currently with this commit fresh throws if no `./deno.json` exists without a helpful error message.

```txt
error: Uncaught (in promise) TypeError: Module not found "file:[PATH]/deno.json".
    at file:[PATH]/fresh.gen.ts:5:20
  await import(entrypoint);
  ^
    at async dev (https://raw.githubusercontent.com/denoland/fresh/main/src/dev/mod.ts:187:3)
    at async file:[PATH]/dev.ts:5:1
```